text/0000-stdlib-prime-proposal.md

```diff
@@ -0,0 +1,144 @@
+```
+**Usage:**
+```pony
+env.out.print(Prime.lcm(3, 6)) // 6
+env.out.print(Prime.lcm(4, 7)) // 28
+```
+* `next_prime(num)`: returns the prime following `num` (not `num` itself if it is prime). This function currently uses the code from `PrimeIterator`.
+```pony
+fun next_prime[A: (Integer[A] val & Unsigned) = USize](num: A): A
+```
+**Usage:**
+```pony
+env.out.print(Prime.next_prime(10)) // 11
+env.out.print(Prime.next_prime(10000)) // 10007
+```
+* `prime_factors(num)`: the most interesting method in my opinion. It returns as an `Array` all the prime factors of a number.
+This method iterates using the `PrimeIterator`, and does not pack reoccurring prime factors into powers of these (this means that `primes_factors(24)` will output `[2; 2; 2; 3]`).
+```pony
+fun prime_factors[A: (Integer[A] val & Unsigned) = USize](num: A): Array[A] ref
+```
+**Usage:**
+```pony
+env.out.print("24 = " + " * ".join(
+  Prime.prime_factors(24).values())
+) // 24 = 2 * 2 * 2 * 3
+```
+
+## `class PrimeIterator`
+
+This class is an `Iterator` listing the prime numbers from `1` or a given number to a limit (by default the maximum value).
+The prime following the limit will be included in the Iterator's values.
+
+* `new create(limit)`: creates a `PrimeIterator`, the `limit` being by default the maximum value of the type
+* `new start_at(last, limit)`: creates a `PrimeIterator`, with the given starting value (`last`)
+* `has_next()`: tells if the last prime number spitted out is below the limit
+* `next()`: gives you the next prime number
+
+**Usage:**
+
+```pony
+// prints all the prime numbers from 1 to 100
+for prime in PrimeIterator do
+  env.out.print(prime.string())
+end
+// will output:
+// 2
+// 3
+// ...
+// 101
+```
+
+# How We Teach This
+
+As this utilities will be in the `math` package for the standard library, their explanations should be done within the `stdlib` documentation.
+
+Mathematical yet easy to understand and to apprehend terminology should be used.
+Words like `coprimes`, `gcm` and `lcm` could have links to articles explaining them.
+
+If a tutorial for the `math` package comes up, this feature could be introduced or evoqued, but I do not feel there is a need to.
+
+# How We Test This
+
+Unit testing can easily be implemented: a good part of them can already be found in the `/test` directory of [pony-primes](https://github.com/adri326/pony-primes).
+
+The algorithms used here are common and so can easily be fixed if they happen to break.
+
+None of the code here have an impact on the other parts of the stdlib and only the `builtin` library is used.
+
+# Drawbacks
+
+This code would sit next to fibonacci's code and should not require much maintenance (the fibonacci code hasn't been updated in 2 years).
+
+However, this feature is not important an important one, and only implements common algorithms which are rarely implemented in other languages' standard library.
+It could thus find a better home in `main.actor`.
```

Comment:
You can update this to say that it could remain as an independent library. `main.actor` is going to be a documentation hosting site and package discovery site.

Reply:
Done!